danascheider/skyrim_inventory_management_frontend
# Enable user to create inventory list item from the front end

## Context

[**Enable user to create inventory list item from the front end**](https://trello.com/c/Mm9Fq6W2/149-enable-user-to-create-inventory-list-item-from-the-front-end)

Currently, users can view, create, update, and destroy inventory lists in the UI, but they aren't able to add, remove, or edit items on those lists. We want users to be able to add list items to inventory lists in the UI. It should not be possible to add an item to an aggregate list this way.

## Changes

* Add `createInventoryListItem` function to `simApi` to make requests to create a list item
* Add `performInventoryListItemCreate` function to `InventoryListsProvider` to make requests and handle responses from the server
* `InventoryListItemCreateForm` component
* Modify `ShoppingListItemCreateForm` to have a `Unit Weight` field
* Add tests for creating a new inventory list item, including happy path and error cases
* Add `performInventoryListItemCreate` function to the docs on the inventory lists context

## Manual Test Cases

In addition to testing creation of inventory list items, make sure you can create shopping list items and the unit weight is handled correctly.

## Screenshots and GIFs

### 1201px

![issue-149_1201px](https://user-images.githubusercontent.com/5115928/129501865-b1052d8d-559d-4e4f-9432-0cf08540c599.png)

### 1025px
![issue-149_769px](https://user-images.githubusercontent.com/5115928/129501898-e2b11681-e0e0-4767-b87b-848234833e37.png)

### 769px

![issue-149_1025px](https://user-images.githubusercontent.com/5115928/129501880-5734ee57-1138-4847-b1e6-ed0b718be586.png)

### 601px

![issue-149_601px](https://user-images.githubusercontent.com/5115928/129501914-423137f8-ebf1-4dde-abdd-40a53143ee28.png)

### 425px

![issue-149_425px](https://user-images.githubusercontent.com/5115928/129501920-1228e876-3022-494c-bdbc-9b08fc134e36.png)
